### ISSUE TITLE
Prettier ignores "ignore" rule

### ISSUE THREAD

**Issue by rowild** [prettier/prettier#5602]
Adding 
`<!-- prettier-ignore -->`
to an ejs file is ignored.

Please provide a section in the docu for how to do that correctly, if this way is the wrong way. Thank you!

**Comment by lydell** [comment]
Hi! Can you reproduce this in the [playground](http://prettier.io/playground/), click the "Copy markdown" button and paste the output here, please?

**Comment by rowild** [comment]
I have no idea how to reproduce that in playground. May I ask you to help me out with this, please?

I have something like this in my `index.ejs`:

```ejs
<!-- prettier-ignore -->
<%- include ./src/partials/shared/_data-header %>

<%- include ./src/partials/shared/header %>

<%- include ./src/partials/shared/main-menu %>
```

In VSCode, the result is

```ejs
<!-- prettier-ignore -->
<%- include ./src/partials/shared/_data-header %> <%- include ./src/partials/shared/header %> <%- include ./src/partials/shared/main-menu %>
```

My `.prettierrc`:

```json
{
  "trailingComma": "es5",
  "tabWidth": 2,
  "printWidth": 180,
  "semi": false,
  "singleQuote": true,
  "prettier.disableLanguages": [
    "html",
    "ejs",
    "javascript",
    "json",
    "css"
  ]
}
```

I need to add the `index.ejs` file to `.prettierignore` in order to prevent automatic formatting.

Thank you!

**Comment by rowild** [comment]
1. How would I test a "ignore" functionality on the CLI??

2. When I deactivate Prettier or put  `index.ejs` into `.prettierignore`, VSCode formats as expected, so there are no other extension interfering. 
(But even though: of course I tested with all extensions turned off  before posting!)

Did you test in playground? If so, how? Please let me know!

**Comment by rowild** [comment]
Again - the question is: what is the proper configuration to advice `prettier` to ignore ejs files?

If you have one, please post or - even better - add to the Readme.

If prettier can't handle `<!-- prettier-ignore -->` in `ejs` files, that has to be mentioned somewhere, too.

**Comment by rowild** [comment]
Command line: `prettier index.ejs` returns:

```
[warn] Ignored unknown option { disableLanguages: ["html", "ejs", "js", "javascript", "json", "css"] }.
[error] No parser could be inferred for file: index.ejs
```

What is wrong with "disableLanguages"???

**Comment by lydell** [comment]
I think you want make a file called `.prettierignore` with the following contents?

```
*.ejs
```

.prettierignore works like .gitignore

**Comment by rowild** [comment]
@lydell sorry to vote you down, but my `.prettierignore` is working, as mentioned above in the context of adding index.ejs. 
(What is not working is "disabledLanguages" in `.prettierrc`, even though mentioned at eom other place... but that would be a separate bug report...)

What is NOT working is  -  again  -  the commenting out via `<!-- prettier-ignore -->`

I expect prettier to either be able to handle that comment and not format anything, or - if prettier doesn't know anything about ejs - leave the file completely alone and not make a mess of it. Or add proper documentation how to handle ejs files (and pug, handlebars...).

I hope I am clear now. Seems to be a confusing topic for everybody...

**Comment by hawkrives** [comment]
@rowild Can you reproduce your issue on the playground? https://prettier.io/playground/ (there's a "Copy Markdown" button that will give you a comment to paste back here.)

**Comment by hawkrives** [comment]
@rowild Yes, I did. I did not see an issue template, so I requested one. 

I do not seem to be able to reproduce your problem: https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhAWkwAgA4BOcMMAlnAZqQOZQRE7YB8AOlKgKS6lRgA2AVwAmcHADoA9AGcCYCXgCGBMgr5TpACyVwhEgPpCFMBZg1wFIgjg6sobDNnxES5SjToNm9rjh79hopIycorKpKrqUlpEumYWFNa29li4hMRkFFS09KJe7D5+giLi0rLySipqmtq6ALYKPJi1CAKJIAA0IBB4ZNBSyKBKBBAA7gAK2lD9KAoAbhCkQh0gAEYECmAA1sQAyopgPNTIMAQCcJ0aMLV8AOoapPBS+3A7CFIPpLMPAJ7I4FL9To8KQUGBjdbUerIABmEXOIAAVlIAB4AIXWW12CmaABkeHAYXDOkjkTtDnw4ABFAQQeCEtTw0Iggh-S7XZaEHgwG6LGAaZAADgADJ1CBAQTd1ng-mlmbMCZ0iABHASkIjghSQhT0kGdEG1UjHU7w95QagU6m0glIWEMzrGFY8oR85AAJnt61IfEOAGEILUoSg6FAFSABCCACoKFbTW0ggC+8aAA

**Comment by hawkrives** [comment]
Additionally, once https://github.com/prettier/prettier/issues/5557 is fixed, you should be able to do 

```
<!-- prettier-ignore-start -->
<%- include ./src/partials/shared/_data-header %>

<%- include ./src/partials/shared/header %>

<%- include ./src/partials/shared/main-menu %>
<!-- prettier-ignore-end -->
```

**Comment by rowild** [comment]
@hawkrives Thanks for posting your link! You use "html" as parser... so I assume html parser has to be used for ejs? Above was mentioned "markdown"... Sorry, but the playground is just not really very helpful compared to others like scss e.g...

However, what becomes clear is that I have to put the ignore line above EVERY SINGLE ejs import.

Looking at the documentation again, this is simply NOT CLEAR.
https://prettier.io/docs/en/ignore.html

Are you a core dev of prettier?

**Comment by rowild** [comment]
@hawkrives Thanks a lot!

- [x] Your issue is that  in a .ejs file is ignored when you save your file, since you use the VSCode Prettier extension.
- [x]   If you put  before each line that you're trying to ignore, it works?
- [ ]   As far as I can tell, Prettier doesn't even format .ejs files by default – see #5474 (comment).
- []   If you disable your VSCode plugin, save the file with the correct formatting, and then run prettier from the command line, does it still reformat it incorrectly?

Ad 3: In my case, VSCode prettier does format the code. I know this, because when I have ALL extensions disabled, no reformatting at all happens (meaning: there is not even an formatting operation by VS Code). When I activate ONLY prettier, it does execute some formatting (correct formatting, when prettier ignore tags are used correctly).
It must be said that there is a variety of prettier plugins available or VS Code - and I only tested the first one (with the most votings) called "Prettier - Code formatter" by Esben Petersen. Who knows - maybe other plugins work differently. But when you look at [4] below, we can now safely assume that this is a problem that only appears within VS Code.

Ad 4: (... I assume you mean disable the "prettier VS Code plugin"): 
My scenario:  Prettier is installed globally (/usr/local/bin/prettier). Formatting the index.ejs with `prettier --no-config index.ejs` returns this error:
`[error] No parser could be inferred for file: index.ejs`
Tested from within VS Code's terminal and iterms2. 
This might be the correct behaviour...

Well, I know my way around thanks to you! 
This issue could be closed.
